debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisDatabaseHistory.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    @Override
+    public void configure(Configuration config, HistoryRecordComparator comparator, DatabaseHistoryListener listener, boolean useCatalogBeforeSchema) {
+        if (!config.validateAndRecord(ALL_FIELDS, logger::error)) {
+            throw new ConnectException(
+                    "Error configuring an instance of " + getClass().getSimpleName() + "; check the logs for details");
+        }
+        config.validateAndRecord(ALL_FIELDS, logger::error);
+        this.config = config;
+        // fetch the properties. as a fallback, if we did not specify the redis address, we will take it and all the credentials from the sink
+        this.address = this.config.getString(PROP_ADDRESS.name());
+        if (this.address == null) {
+            // try to take the connection details from the redis sink
+            this.address = this.config.getString(SINK_PROP_PREFIX + "address");
+            this.user = this.config.getString(SINK_PROP_PREFIX + "user");
+            this.password = this.config.getString(SINK_PROP_PREFIX + "password");
+        }
+        else {
+            this.user = this.config.getString(PROP_USER.name());
+            this.password = this.config.getString(PROP_PASSWORD.name());
+        }
+        this.redisKeyName = this.config.getString(PROP_KEY);
+        LOGGER.info("rediskeyname:" + this.redisKeyName);
+        // load retry settings
+        this.initialRetryDelay = this.config.getInteger(PROP_RETRY_INITIAL_DELAY);
+        this.maxRetryDelay = this.config.getInteger(PROP_RETRY_MAX_DELAY);
+
+        super.configure(config, comparator, listener, useCatalogBeforeSchema);
+    }
+
+    @Override
+    public synchronized void start() {
+        super.start();
+        LOGGER.info("Starting RedisDatabaseHistory");
+        this.connect();
+    }
+
+    @Override
+    protected void storeRecord(HistoryRecord record) throws DatabaseHistoryException {
+        if (record == null) {
+            return;
+        }
+        try {
+            String line = writer.write(record.document());
+            Uni.createFrom().item(() -> {
+                return (StreamEntryID) client.xadd(this.redisKeyName, (StreamEntryID) null, Collections.singletonMap("schema", line));
+            })
+                    .onFailure().invoke(
+                            f -> {
+                                LOGGER.warn("Writing to database history stream failed with " + f);
+                                LOGGER.warn("Will retry");
+                            })
+                    .onFailure(JedisConnectionException.class).invoke(
+                            f -> {
+                                LOGGER.warn("Attempting to reconnect to redis ");
+                                this.connect();
+                            })
+                    // retry on failure with backoff
+                    .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
```

Comment:
Instead of introducing Mutiny as a new dependency, can't we use Debezium's own `DelayStrategy` here?

Reply:
I also used those in the OffsetStore so wanted to be consistent. I thought Mutiny was baked into Quarkus and since this was very readable thought it was the better way to implement. In fact, was planning on aligning the RedisStreamConsumer to handle the same way in the near future. When I tested it with the RedisOffsetStore it seemed to work well. Do you see an issue using it?

Reply:
I see, I may have missed this for the offset storage. I'd agree from a purely Debezium Server perspective, but we'd also want to enable usage of these alternative history/offset implementations with the embedded engine, and for that context we should be a bit more conservative when it comes to adding more dependencies. Now I'm not pushing back wholesale on adding Mutiny, if we actually perceive there's significant value add for our purposes. But given we do have an existing mechanism in Debezium in form of the `DelayStrategy`, I'm not convinced yet the additional dependency is worth it. WDYT?

Reply:
oh, I see. I'll look into it on Sunday and try to provide a change before Monday morning.

Reply:
Ok, cool, thanks!

@jpechane, let's try and get this one in for the CR1 on Wed/Thu.